### ISSUE TITLE
MySQL: double quoted string literals are not concatenated

### ISSUE THREAD

**Issue by yoichi** [sqlfluff/sqlfluff#3870]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

https://dev.mysql.com/doc/refman/8.0/en/string-literals.html
Quoted string literals are concatenated as:

```
mysql> SELECT 'foo' 'bar';
+--------+
| foo    |
+--------+
| foobar |
+--------+
1 row in set (0.00 sec)

mysql> SELECT "foo" "bar";
+--------+
| foo    |
+--------+
| foobar |
+--------+
1 row in set (0.00 sec)

mysql> SELECT 'foo' "bar";
+--------+
| foo    |
+--------+
| foobar |
+--------+
1 row in set (0.00 sec)

mysql> SELECT "foo" 'bar';
+--------+
| foo    |
+--------+
| foobar |
+--------+
1 row in set (0.00 sec)
```

tested with MySQL 8.0.29. But queries double quoted string literals causes parse error on `sqlfluff lint`.

### Expected Behaviour

Do not cause parse error.

### Observed Behaviour

Cause parse error.

### How to reproduce

```
$ cat concat.sql
SELECT "foo" "bar";
SELECT 'foo' "bar";
SELECT "foo" 'bar';
```

```
$ sqlfluff lint --dialect=mysql concat.sql
== [concat.sql] FAIL
L:   1 | P:  14 |  PRS | Line 1, Position 14: Found unparsable section: '"bar"'
L:   2 | P:   8 | L064 | Inconsistent use of preferred quote style 'double
                       | quotes'. Use "foo" instead of 'foo'.
L:   2 | P:  14 |  PRS | Line 2, Position 14: Found unparsable section: '"bar"'
L:   3 | P:  14 |  PRS | Line 3, Position 14: Found unparsable section: "'bar'"
WARNING: Parsing errors found and dialect is set to 'mysql'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Dialect

MySQL

### Version

sqlfluff, version 1.3.1

### Configuration

no configuration file

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
